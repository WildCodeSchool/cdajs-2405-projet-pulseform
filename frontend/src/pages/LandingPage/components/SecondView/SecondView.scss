 @import "@assets/variables.scss";

 .second-view {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100vh;

    &__img-container {
      display: none;

      @media screen and (min-width: 500px) {
        display: block;
        width: 100%;
        height: 58vh;
        overflow: hidden;
        box-shadow: inset 0 0 20px 6px rgba(0, 0, 0, 0.2);

        &__img {
          width: 100%;
        }
      }
    }

    &__description {
      @include h1-phone;
      color: $primary;
      text-align: center;
      margin: 44px 16px 43px;

      @media screen and (min-width: 500px) {
        margin: 24px 0 30px;
      }
    }

    &__step {
      display: flex;
      flex-direction: column;
      gap: 34px;

      @media screen and (min-width: 500px) {
        flex-direction: row;
        justify-content: space-around;
        gap: 37px;
      }

      &__content {
        @include h2-phone;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        color: $white;
        background: $primary;
        border-radius: 10px;
        width: 178px;
        height: 150px;
        padding: 0 16px;
        text-align: center;
        z-index: 1;

        @media screen and (min-width: 500px) {
          &:first-child {
            margin-top: 62px;
          }

          &:last-child {
            margin-top: 62px;
          }
        }
      }
    }

    &__blob {
      position: absolute;
      right: 0;
      left: 0;
      bottom: -100vh;
      height: 124px;
      z-index: 0;
      overflow: hidden;

      @media screen and (min-width: 500px) {
        height: 200px;
        left: 30vw;
      }

      &__img {
        width: 440px;
        height: 360px;

        @media screen and (min-width: 500px) {
          width: 644px;
          height: 528px;
        }
      }
    }

    &__buttom {
      position: absolute;
      display: flex;
      justify-content: center;
      z-index: 1;
      bottom: -94%;
      width: 100%;

      @media screen and (min-width: 500px) {
        bottom: -97%;
      }
    }
  }