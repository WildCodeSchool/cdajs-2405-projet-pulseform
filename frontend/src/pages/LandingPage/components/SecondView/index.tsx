import BasicButton from "@components/atoms/BasicButton";

import blobSecondView from "@assets/blobs/blob2.svg";
import Programs from "@assets/images/programs.svg";
import { useTranslation } from "react-i18next";
import "./SecondView.scss";

function SecondView() {
  const { t } = useTranslation();

  return (
    <div className="second-view">
      <div className="second-view__img-container">
        <img
          className="second-view__img-container__img"
          src={Programs}
          alt="programs"
        />
      </div>
      <p className="second-view__description">{t("LANDING_DESCRIPTION_2")}</p>
      <div className="second-view__step">
        <p className="second-view__step__content">{t("PROGRAMS_TO_DICOVER")}</p>
        <p className="second-view__step__content">
          {t("FOLLOW_YOUR_PROGRESS")}
        </p>
        <p className="second-view__step__content">
          {t("PERSONALIZED_COACHING")}
        </p>
      </div>
      <div className="second-view__buttom">
        <BasicButton>{t("TEST_PROGRAM")}</BasicButton>
      </div>
      <div className="second-view__blob">
        <img
          className="second-view__blob__img"
          src={blobSecondView}
          alt="blob"
        />
      </div>
    </div>
  );
}

export default SecondView;
