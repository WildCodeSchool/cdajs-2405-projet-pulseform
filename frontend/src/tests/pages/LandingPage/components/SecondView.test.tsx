import SecondView from "@pages/LandingPage/components/SecondView";
import { render } from "@testing-library/react";

describe("SecondView", () => {
  it("renders the SecondView component", () => {
    const tree = render(<SecondView />);
    expect(tree).toMatchSnapshot();
  });
});
