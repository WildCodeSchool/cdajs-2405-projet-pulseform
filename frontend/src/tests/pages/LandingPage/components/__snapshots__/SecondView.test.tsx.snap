// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SecondView > renders the SecondView component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="second-view"
      >
        <div
          class="second-view__img-container"
        >
          <img
            alt="programs"
            class="second-view__img-container__img"
            src="/src/assets/images/programs.svg"
          />
        </div>
        <p
          class="second-view__description"
        >
          LANDING_DESCRIPTION_2
        </p>
        <div
          class="second-view__step"
        >
          <p
            class="second-view__step__content"
          >
            PROGRAMS_TO_DICOVER
          </p>
          <p
            class="second-view__step__content"
          >
            FOLLOW_YOUR_PROGRESS
          </p>
          <p
            class="second-view__step__content"
          >
            PERSONALIZED_COACHING
          </p>
        </div>
        <div
          class="second-view__buttom"
        >
          <button
            class="basic-button basic-button--basic"
            type="button"
          >
            TEST_PROGRAM
          </button>
        </div>
        <div
          class="second-view__blob"
        >
          <img
            alt="blob"
            class="second-view__blob__img"
            src="/src/assets/blobs/blob2.svg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="second-view"
    >
      <div
        class="second-view__img-container"
      >
        <img
          alt="programs"
          class="second-view__img-container__img"
          src="/src/assets/images/programs.svg"
        />
      </div>
      <p
        class="second-view__description"
      >
        LANDING_DESCRIPTION_2
      </p>
      <div
        class="second-view__step"
      >
        <p
          class="second-view__step__content"
        >
          PROGRAMS_TO_DICOVER
        </p>
        <p
          class="second-view__step__content"
        >
          FOLLOW_YOUR_PROGRESS
        </p>
        <p
          class="second-view__step__content"
        >
          PERSONALIZED_COACHING
        </p>
      </div>
      <div
        class="second-view__buttom"
      >
        <button
          class="basic-button basic-button--basic"
          type="button"
        >
          TEST_PROGRAM
        </button>
      </div>
      <div
        class="second-view__blob"
      >
        <img
          alt="blob"
          class="second-view__blob__img"
          src="/src/assets/blobs/blob2.svg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
